data/src/client.rs

```diff
@@ -1175,17 +1201,19 @@ pub enum WhoStatus {
 }
 
 /// Group channels together into as few JOIN messages as possible
-fn group_joins(config: &config::Server) -> impl Iterator<Item = proto::Message> + '_ {
+fn group_joins<'a>(
+    config: &'a config::Server,
+    channels: &'a [String],
```

Comment:
Can we update this to take keys instead of the entire config?

It's confusing (and a code smell) we pass in self.config and self.config.channels.

Reply:
lets do `channels: &'a [String], keys: &'a [String]`

Reply:
An artifact of the first implementation that I neglected to revisit.  You're absolutely right of course, the interface doesn't really make sense.  Thank you for catching it!  :)